Index: gcc-7-7.2.0-12.1/src/libgo/go/archive/tar/stat_atim.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/archive/tar/stat_atim.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/archive/tar/stat_atim.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build linux dragonfly openbsd solaris
+// +build gnu linux dragonfly openbsd solaris
 
 package tar
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/cmd/go/signal_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/cmd/go/signal_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/cmd/go/signal_unix.go
@@ -1,8 +1,9 @@
+
 // Copyright 2012 The Go Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package main
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/crypto/x509/root_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/crypto/x509/root_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/crypto/x509/root_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package x509
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/fd_poll_runtime.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/fd_poll_runtime.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/fd_poll_runtime.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd windows solaris
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd windows solaris
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/interface_stub.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/interface_stub.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/interface_stub.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build nacl
+// +build nacl gnu
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/internal/socktest/switch_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/internal/socktest/switch_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/internal/socktest/switch_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package socktest
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/port_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/port_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/port_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris nacl
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris nacl
 
 // Read system port mappings from /etc/services
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/dir_largefile.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/dir_largefile.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/dir_largefile.go
@@ -5,7 +5,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build linux solaris,386 solaris,sparc
+// +build gnu linux solaris,386 solaris,sparc
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/dir_regfile.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/dir_regfile.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/dir_regfile.go
@@ -5,6 +5,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// +build !gnu
 // +build !linux
 // +build !solaris !386
 // +build !solaris !sparc
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/dir_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/dir_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/dir_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/exec_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/exec_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/exec_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/stat_atim.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/stat_atim.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/stat_atim.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build linux openbsd solaristag
+// +build gnu linux openbsd solaristag
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/stat.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/stat.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/stat.go
@@ -4,6 +4,7 @@
 
 // +build !darwin
 // +build !freebsd
+// +build !gnu
 // +build !linux
 // +build !netbsd
 // +build !openbsd
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/stat_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/stat_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/stat_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/sys_uname.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/sys_uname.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/sys_uname.go
@@ -4,7 +4,7 @@
 
 // For systems which only store the hostname in uname (Solaris).
 
-// +build solaris irix rtems
+// +build gnu solaris irix rtems
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/user/listgroups_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/user/listgroups_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/user/listgroups_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build dragonfly darwin freebsd !android,linux netbsd openbsd
+// +build dragonfly darwin freebsd gnu !android,linux netbsd openbsd
 
 package user
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/wait_unimp.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/wait_unimp.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/wait_unimp.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build dragonfly nacl netbsd openbsd solaris
+// +build dragonfly gnu nacl netbsd openbsd solaris
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/path/filepath/path_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/path/filepath/path_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/path/filepath/path_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package filepath
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/runtime/env_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/runtime/env_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/runtime/env_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 package runtime
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/runtime/lock_sema.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/runtime/lock_sema.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/runtime/lock_sema.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin nacl netbsd openbsd plan9 solaris windows
+// +build darwin gnu nacl netbsd openbsd plan9 solaris windows
 
 package runtime
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/runtime/netpoll.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/runtime/netpoll.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/runtime/netpoll.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 package runtime
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/runtime/signal_gccgo.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/runtime/signal_gccgo.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/runtime/signal_gccgo.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris
 
 package runtime
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/runtime/signal_sighandler.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/runtime/signal_sighandler.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/runtime/signal_sighandler.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package runtime
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/syscall/errstr.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/syscall/errstr.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/syscall/errstr.go
@@ -4,6 +4,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// +build !gnu
 // +build !linux
 
 package syscall
Index: gcc-7-7.2.0-12.1/src/libgo/go/syscall/libcall_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/syscall/libcall_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/syscall/libcall_posix.go
@@ -9,6 +9,8 @@
 // Note that sometimes we use a lowercase //sys name and
 // wrap it in our own nicer implementation.
 
+// +build !gnu
+
 package syscall
 
 import "unsafe"
Index: gcc-7-7.2.0-12.1/src/libgo/go/syscall/libcall_posix_largefile.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/syscall/libcall_posix_largefile.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/syscall/libcall_posix_largefile.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build linux solaris,386 solaris,sparc
+// +build gnu linux solaris,386 solaris,sparc
 
 // POSIX library calls on systems which use the largefile interface.
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/syscall/libcall_posix_regfile.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/syscall/libcall_posix_regfile.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/syscall/libcall_posix_regfile.go
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// +build !gnu
 // +build !linux
 // +build !solaris !386
 // +build !solaris !sparc
Index: gcc-7-7.2.0-12.1/src/libgo/go/syscall/libcall_posix_utimesnano.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/syscall/libcall_posix_utimesnano.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/syscall/libcall_posix_utimesnano.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd openbsd netbsd solaris
+// +build darwin dragonfly freebsd gnu openbsd netbsd solaris
 
 // General POSIX version of UtimesNano.
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/cgo_resnew.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/cgo_resnew.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/cgo_resnew.go
@@ -3,7 +3,7 @@
 // license that can be found in the LICENSE file.
 
 // +build cgo,!netgo
-// +build darwin linux,!android netbsd solaris
+// +build darwin gnu linux,!android netbsd solaris
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/cgo_sockold.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/cgo_sockold.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/cgo_sockold.go
@@ -3,7 +3,7 @@
 // license that can be found in the LICENSE file.
 
 // +build cgo,!netgo
-// +build darwin dragonfly freebsd netbsd openbsd
+// +build darwin dragonfly freebsd gnu netbsd openbsd
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/hook_cloexec.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/hook_cloexec.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/hook_cloexec.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build freebsd linux
+// +build freebsd gnu linux
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/sock_cloexec.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/sock_cloexec.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/sock_cloexec.go
@@ -5,7 +5,7 @@
 // This file implements sysSocket and accept for platforms that
 // provide a fast path for setting SetNonblock and CloseOnExec.
 
-// +build freebsd linux
+// +build freebsd gnu linux
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/sockoptip_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/sockoptip_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/sockoptip_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd windows
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd windows
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/syscall/exec_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/syscall/exec_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/syscall/exec_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris
 
 // Fork, exec, wait, etc.
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/exec/lp_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/exec/lp_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/exec/lp_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package exec
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/signal/signal_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/signal/signal_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/signal/signal_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 package signal
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/error_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/error_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/error_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/file_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/file_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/file_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/path_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/path_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/path_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/sys_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/sys_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/sys_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build dragonfly linux netbsd openbsd solaris
+// +build dragonfly gnu linux netbsd openbsd solaris
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/user/decls_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/user/decls_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/user/decls_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd
 // +build cgo
 
 package user
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/user/lookup_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/user/lookup_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/user/lookup_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd !android,linux netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu !android,linux netbsd openbsd solaris
 // +build cgo
 
 package user
Index: gcc-7-7.2.0-12.1/src/libgo/go/syscall/env_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/syscall/env_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/syscall/env_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 // Unix environment variables.
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/syscall/exec_bsd.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/syscall/exec_bsd.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/syscall/exec_bsd.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu netbsd openbsd solaris
 
 package syscall
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/syscall/sockcmsg_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/syscall/sockcmsg_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/syscall/sockcmsg_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris
 
 // Socket control messages
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/syscall/syscall_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/syscall/syscall_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/syscall/syscall_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris
 
 package syscall
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/time/sys_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/time/sys_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/time/sys_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package time
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/time/zoneinfo_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/time/zoneinfo_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/time/zoneinfo_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin,386 darwin,amd64 dragonfly freebsd linux,!android nacl netbsd openbsd solaris
+// +build darwin,386 darwin,amd64 dragonfly freebsd gnu linux,!android nacl netbsd openbsd solaris
 
 // Parse "zoneinfo" time zone file.
 // This is a fairly standard file format used on OS X, Linux, BSD, Sun, and others.
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/addrselect.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/addrselect.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/addrselect.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris
 
 // Minimal RFC 6724 address selection.
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/conf.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/conf.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/conf.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/dnsclient_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/dnsclient_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/dnsclient_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris
 
 // DNS client: see RFC 1035.
 // Has to be linked into package net for Dial.
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/dnsconfig_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/dnsconfig_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/dnsconfig_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris
 
 // Read system DNS config from /etc/resolv.conf
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/fd_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/fd_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/fd_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/fd_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/fd_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/fd_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/file_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/file_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/file_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/hook_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/hook_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/hook_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/iprawsock_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/iprawsock_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/iprawsock_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/ipsock_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/ipsock_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/ipsock_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/lookup_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/lookup_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/lookup_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/nss.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/nss.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/nss.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/sockopt_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/sockopt_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/sockopt_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris windows
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/sock_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/sock_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/sock_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/tcpsockopt_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/tcpsockopt_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/tcpsockopt_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris windows
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/tcpsockopt_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/tcpsockopt_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/tcpsockopt_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build freebsd linux netbsd
+// +build freebsd gnu linux netbsd
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/tcpsock_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/tcpsock_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/tcpsock_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/udpsock_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/udpsock_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/udpsock_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/unixsock_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/unixsock_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/unixsock_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/exec_posix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/exec_posix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/exec_posix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/file_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/file_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/file_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/os/executable_procfs.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/os/executable_procfs.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/os/executable_procfs.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build linux netbsd openbsd dragonfly nacl
+// +build gnu linux netbsd openbsd dragonfly nacl
 
 package os
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/syscall/timestruct.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/syscall/timestruct.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/syscall/timestruct.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package syscall
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/cgo_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/cgo_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/cgo_unix.go
@@ -3,7 +3,7 @@
 // license that can be found in the LICENSE file.
 
 // +build cgo,!netgo
-// +build darwin dragonfly freebsd linux netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux netbsd openbsd solaris
 
 package net
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/archive/tar/stat_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/archive/tar/stat_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/archive/tar/stat_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build linux darwin dragonfly freebsd openbsd netbsd solaris
+// +build gnu linux darwin dragonfly freebsd openbsd netbsd solaris
 
 package tar
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/crypto/rand/eagain.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/crypto/rand/eagain.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/crypto/rand/eagain.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package rand
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/crypto/rand/rand_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/crypto/rand/rand_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/crypto/rand/rand_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd plan9 solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd plan9 solaris
 
 // Unix cryptographically secure pseudorandom number
 // generator.
Index: gcc-7-7.2.0-12.1/src/libgo/go/exp/terminal/util.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/exp/terminal/util.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/exp/terminal/util.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build linux
+// +build gnu linux
 
 // Package terminal provides support functions for dealing with terminals, as
 // commonly found on UNIX systems.
Index: gcc-7-7.2.0-12.1/src/libgo/go/mime/type_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/mime/type_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/mime/type_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package mime
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/plugin/plugin_dlopen.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/plugin/plugin_dlopen.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/plugin/plugin_dlopen.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build linux,cgo darwin,cgo
+// +build linux,cgo darwin,cgo gnu,cgo
 
 package plugin
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/syscall/dirent.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/syscall/dirent.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/syscall/dirent.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package syscall
 
Index: gcc-7-7.2.0-12.1/src/libgo/runtime/env_posix.c
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/runtime/env_posix.c
+++ gcc-7-7.2.0-12.1/src/libgo/runtime/env_posix.c
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris windows
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris windows
 
 #include "runtime.h"
 #include "array.h"
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/internal/socktest/sys_unix.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/internal/socktest/sys_unix.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/internal/socktest/sys_unix.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+// +build darwin dragonfly freebsd gnu linux nacl netbsd openbsd solaris
 
 package socktest
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/plugin/plugin_stubs.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/plugin/plugin_stubs.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/plugin/plugin_stubs.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build !linux,!darwin !cgo
+// +build !gnu !linux,!darwin !cgo
 
 package plugin
 
Index: gcc-7-7.2.0-12.1/src/libgo/go/net/internal/socktest/sys_cloexec.go
===================================================================
--- gcc-7-7.2.0-12.1.orig/src/libgo/go/net/internal/socktest/sys_cloexec.go
+++ gcc-7-7.2.0-12.1/src/libgo/go/net/internal/socktest/sys_cloexec.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build freebsd linux
+// +build freebsd gnu linux
 
 package socktest
 
