Index: gcc/testsuite/gcc.target/arm/fpscr.c
===================================================================
--- a/src/gcc/testsuite/gcc.target/arm/fpscr.c	(revision 249596)
+++ a/src/gcc/testsuite/gcc.target/arm/fpscr.c	(nonexistent)
@@ -1,16 +0,0 @@
-/* Test the fpscr builtins.  */
-
-/* { dg-do compile } */
-/* { dg-require-effective-target arm_fp_ok } */
-/* { dg-skip-if "need fp instructions" { *-*-* } { "-mfloat-abi=soft" } { "" } } */
-/* { dg-add-options arm_fp } */
-
-void
-test_fpscr ()
-{
-  volatile unsigned int status = __builtin_arm_get_fpscr ();
-  __builtin_arm_set_fpscr (status);
-}
-
-/* { dg-final { scan-assembler "mrc\tp10, 7, r\[0-9\]+, cr1, cr0, 0" } } */
-/* { dg-final { scan-assembler "mcr\tp10, 7, r\[0-9\]+, cr1, cr0, 0" } } */
Index: gcc/config/arm/arm-builtins.c
===================================================================
--- a/src/gcc/config/arm/arm-builtins.c	(revision 249596)
+++ a/src/gcc/config/arm/arm-builtins.c	(revision 249586)
@@ -1893,10 +1893,10 @@
 	= build_function_type_list (unsigned_type_node, NULL);
 
       arm_builtin_decls[ARM_BUILTIN_GET_FPSCR]
-	= add_builtin_function ("__builtin_arm_get_fpscr", ftype_get_fpscr,
+	= add_builtin_function ("__builtin_arm_ldfscr", ftype_get_fpscr,
 				ARM_BUILTIN_GET_FPSCR, BUILT_IN_MD, NULL, NULL_TREE);
       arm_builtin_decls[ARM_BUILTIN_SET_FPSCR]
-	= add_builtin_function ("__builtin_arm_set_fpscr", ftype_set_fpscr,
+	= add_builtin_function ("__builtin_arm_stfscr", ftype_set_fpscr,
 				ARM_BUILTIN_SET_FPSCR, BUILT_IN_MD, NULL, NULL_TREE);
     }
 
