Index: gcc-7-7.2.0-12.1/src/libgo/go/crypto/x509/root_gnu.go
===================================================================
--- /dev/null
+++ gcc-7-7.2.0-12.1/src/libgo/go/crypto/x509/root_gnu.go
@@ -0,0 +1,10 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package x509
+
+// Possible certificate files; stop after finding one.
+var certFiles = []string{
+	"/etc/ssl/certs/ca-certificates.crt",                // Debian/Ubuntu/Gentoo etc.
+}
